lib/components/search.jsx

```diff
@@ -0,0 +1,202 @@
+  render () {
+    const {
+      isInstalling,
+      isLoading,
+      matches,
+      selectedPackage,
+      terms
+    } = this.state
+
+    return <div>
+      <SearchInput
+        terms={terms}
+        onChange={this.onChangeTerms}
+        onSubmit={this.onSubmit} />
+      <SearchResults
+        isLoading={isLoading}
+        terms={terms}
+        matches={matches}
+        onSelect={this.onSelectPackage} />
+      { isInstalling && selectedPackage
+        ? <InstallingPackage isInstalling={isInstalling} pkg={selectedPackage} />
+        : null
+      }
+    </div>
+  }
+
+  async search (terms) {
+    try {
+      this.setState({
+        isLoading: true,
+        matches: []
+      })
+      const matches = await libnpm.search(terms, {
+        detailed: true,
+        limit: MAX_RESULTS
+      })
+      this.setState({
+        isLoading: false,
+        matches
+      })
+    } catch (err) {
+      // @todo Show error message
+      this.setState({
+        isLoading: false,
+        matches: []
+      })
+    }
+  }
+
+  async install (pkg) {
+    try {
+      this.setState({
+        isInstalling: true,
+        selectedPackage: pkg
+      })
+      // @todo Implement real install
+      await new Promise((resolve, reject) => setTimeout(resolve, 1000))
+      this.setState({ isInstalling: false })
+      this.props.onExit()
+    } catch (err) {
+      // @todo Show error message
+      this.setState({ isInstalling: false })
+    }
+  }
+
+  onChangeTerms (terms) {
+    this.setState({ terms })
+  }
+
+  onSubmit (terms) {
+    this.search(terms)
+  }
+
+  onSelectPackage (pkg) {
+    this.install(pkg)
+  }
+}
+
+const SearchInput = ({ onChange, onSubmit, terms }) => {
+  return <div>
+    <Color grey bold>Find a package: </Color>
+    <TextInput value={terms} onChange={onChange} onSubmit={onSubmit} />
+  </div>
+}
+
+const SearchResults = ({ isLoading, matches, onSelect, terms }) => {
+  if (isLoading) {
+    return <div><Spinner /> <Color green>Searching...</Color></div>
+  }
+
+  if (isLoading === false && terms && (!matches || !matches.length)) {
+    return <Color grey>No matches found</Color>
+  }
+
+  if (isLoading === false && terms && matches && matches.length) {
+    return <div>
+      <Color grey bold>Results:</Color><br />
+      <PackageSelector matches={matches} onSelect={onSelect} />
+    </div>
+  }
+
+  return null
+}
+
+const PackageSelectIndicator = ({ isSelected }) => {
+  if (!isSelected) {
+    return ' ';
+  }
+
+  return <Color green>{`${figures.pointer} `}</Color>
+};
+
+const formatPackageScore = num => Math.round(num * 100)
+
+const PackageItem = ({ isSelected, value }) => {
+  const {
+    package: {
+      name,
+      publisher: { username }
+    },
+    score: {
+      detail: {
+        maintenance,
+        popularity,
+        quality
+      }
+    }
+  } = value
+
+  const m = formatPackageScore(maintenance)
+  const p = formatPackageScore(popularity)
+  const q = formatPackageScore(quality)
```

Comment:
How do you feel about the `m/p/q` formatting? Do you think it's helpful to people to display them like that?

Reply:
@zkat I think it's pretty vague—I definitely didn't know what it was until I put it together after seeing the `sortBy` option in `libnpmsearch`. It's nice having a little more information in a one-liner, but we're definitely limited in what we can fit. Maybe we could change the one-liner to package name, latest version, maybe publishing user? And once `<PackageView />` is in the mix, these handy details can be displayed with a little more context.

What do you think?

Reply:
How about:
```
Search: Foo
Results (maintenance, popularity, quality):
  -> foo@1.2.3 @zkat (50%/60%/75%)
```

Reply:
@zkat Yea, I think a mini legend works great.

Reply:
@zkat OK, so this update is made. The percentages feel a bit squished—I kinda wish there was a way to tab-align them so they aligned into a column—but it's definitely more clear. Gonna dive into `PackageView` integration soon...